public/app/plugins/panel/canvas/components/connections/Connections.tsx

```diff
@@ -323,9 +331,52 @@ export class Connections {
       }
     }
 
-    // Display temporary vertex during drag
-    this.connectionVertexPath?.setAttribute('d', `M${vx1} ${vy1} L${x} ${y} L${vx2} ${vy2}`);
-    this.connectionSVGVertex!.style.display = 'block';
+    // Check if slope before vertex and after vertex is within snapping tolerance
+    let xSnap = x;
+    let ySnap = y;
+    let deleteVertex = false;
+    // Ignore if control key being held
+    if (!event.ctrlKey) {
+      // Check if segment before and after vertex are close to vertical or horizontal
+      const verticalBefore = Math.abs((x - vx1) / (y - vy1)) < CONNECTION_VERTEX_ORTHO_TOLERANCE;
+      const verticalAfter = Math.abs((x - vx2) / (y - vy2)) < CONNECTION_VERTEX_ORTHO_TOLERANCE;
+      const horizontalBefore = Math.abs((y - vy1) / (x - vx1)) < CONNECTION_VERTEX_ORTHO_TOLERANCE;
+      const horizontalAfter = Math.abs((y - vy2) / (x - vx2)) < CONNECTION_VERTEX_ORTHO_TOLERANCE;
+
+      if (verticalBefore) {
+        xSnap = vx1;
+      } else if (verticalAfter) {
+        xSnap = vx2;
+      }
+      if (horizontalBefore) {
+        ySnap = vy1;
+      } else if (horizontalAfter) {
+        ySnap = vy2;
+      }
+
+      if ((verticalBefore || verticalAfter) && (horizontalBefore || horizontalAfter)) {
+        this.scene.selecto!.rootContainer!.style.cursor = 'move';
+      } else if (verticalBefore || verticalAfter) {
+        this.scene.selecto!.rootContainer!.style.cursor = 'col-resize';
+      } else if (horizontalBefore || horizontalAfter) {
+        this.scene.selecto!.rootContainer!.style.cursor = 'row-resize';
+      }
+
+      const angleOverall = calculateAngle(vx1, vy1, vx2, vy2);
+      const angleBefore = calculateAngle(vx1, vy1, x, y);
+      deleteVertex = Math.abs(angleBefore - angleOverall) < CONNECTION_VERTEX_SNAP_TOLERANCE;
+    }
+
+    if (deleteVertex) {
+      // Display temporary vertex removal
+      this.scene.selecto!.rootContainer!.style.cursor = 'not-allowed';
```

Comment:
is the cursor style change there to indicate that the vertexes are being removed? I'm not sure this is necessary / might be confusing to end user as it seems like they are doing an action that isn't allowed or something 🤔 

https://github.com/grafana/grafana/assets/22381771/7ef8216d-b7b8-484d-8ade-e69b34fc9fcb

Reply:
I want to agree here, feels like this cursor doesn't quite match up with the outcome expected. Maybe would be better to have a treatment for the line or something like that?

Reply:
Agreed, appreciate having more eyes on this!

Reply:
It's subtle, but by simply hiding the vertex during "deletion snap", I think is good enough for now.

![Mar-20-2024 23-29-20](https://github.com/grafana/grafana/assets/60050885/6ffb5caf-00f1-449b-9ab7-3317a7e718db)

Reply:
:shipit: